### PR TITLE
DOC: Remove 1.2 versionadded and changed

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
```
023-04-26T18:50:42.6547459Z /home/runner/work/pandas/pandas/pandas/io/excel/_base.py:docstring of pandas.io.excel._base.read_excel:89: WARNING: Bullet list ends without a blank line; unexpected unindent.
2023-04-26T18:50:42.6548203Z /home/runner/work/pandas/pandas/pandas/io/excel/_base.py:docstring of pandas.io.excel._base.read_excel:90: WARNING: Block quote ends without a blank line; unexpected unindent.
```

**Comment by mroeschke** [comment]
The same errors in my last comment are appearing

**Review by datapythonista — APPROVED** [review]
Looks good other than the CI problems.

I don't fully understand why we want to remove the indentation of the versionchanged in the Excel bullet points, but I trust you've checked the render and looks fine.

Out of curiosity, do we have a written policy for when those versionchanged/versionadded are removed?

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -157,23 +157,18 @@
...
-        The engine `xlrd <https://xlrd.readthedocs.io/en/latest/>`_
-        now only supports old-style ``.xls`` files.
-        When ``engine=None``, the following logic will be
-        used to determine the engine:
+    * If ``path_or_buffer`` is an OpenDocument format (.odf, .ods, .odt),
+      then `odf <https://pypi.org/project/odfpy/>`_ will be used.
+    * Otherwise if ``path_or_buffer`` is an xls format,
+     ``xlrd`` will be used.
+    * Otherwise if ``path_or_buffer`` is in xlsb format,
+      ``pyxlsb`` will be used.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:165]: These first two lines are a lead in to the bullet points that follow.

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -1448,30 +1439,18 @@ class ExcelFile:
...
-           - Otherwise if ``path_or_buffer`` is an xls format,
-             ``xlrd`` will be used.
-           - Otherwise if ``path_or_buffer`` is in xlsb format,
-             `pyxlsb <https://pypi.org/project/pyxlsb/>`_ will be used.
+        * If ``path_or_buffer`` is an OpenDocument format (.odf, .ods, .odt),
+          then `odf <https://pypi.org/project/odfpy/>`_ will be used.
+        * Otherwise if ``path_or_buffer`` is an xls format,
+          ``xlrd`` will be used.
+        * Otherwise if ``path_or_buffer`` is in xlsb format,
+          `pyxlsb <https://pypi.org/project/pyxlsb/>`_ will be used.
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:1447]: Same as above

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/io/excel/_base.py [hunk 3]
```diff
@@ -157,23 +157,18 @@
     - "odf" supports OpenDocument file formats (.odf, .ods, .odt).
     - "pyxlsb" supports Binary Excel files.
 
-    .. versionchanged:: 1.2.0
-        The engine `xlrd <https://xlrd.readthedocs.io/en/latest/>`_
-        now only supports old-style ``.xls`` files.
-        When ``engine=None``, the following logic will be
-        used to determine the engine:
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:164]: I think these two lines should be left in.

File: pandas/io/excel/_base.py [hunk 4]
```diff
@@ -1448,30 +1439,19 @@ class ExcelFile:
         - ``odf`` supports OpenDocument file formats (.odf, .ods, .odt).
         - ``pyxlsb`` supports Binary Excel files.
 
-        .. versionchanged:: 1.2.0
-
-           The engine `xlrd <https://xlrd.readthedocs.io/en/latest/>`_
-           now only supports old-style ``.xls`` files.
-           When ``engine=None``, the following logic will be
-           used to determine the engine:
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:1456]: Same here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks like the change to indent is causing warnings in the docs build for io.excel.

**Comment by mroeschke** [comment]
Feel free to reopen when you have time to circle back here

**Pull request closed without merging**